frontend/src/metabase/models/containers/NewModelOptions/NewModelOptions.tsx

```diff
@@ -14,35 +15,31 @@ import { NoDatabasesEmptyState } from "metabase/reference/databases/NoDatabasesE
 import { getHasDataAccess, getHasNativeWrite } from "metabase/selectors/data";
 import { getSetting } from "metabase/selectors/settings";
 import { getShowMetabaseLinks } from "metabase/selectors/whitelabel";
-import type Database from "metabase-lib/v1/metadata/Database";
 
 import {
+  EducationalButton,
   OptionsGridItem,
   OptionsRoot,
-  EducationalButton,
 } from "./NewModelOptions.styled";
 
 const EDUCATIONAL_LINK = MetabaseSettings.learnUrl("data-modeling/models");
 
 interface NewModelOptionsProps {
-  databases?: Database[];
   location: Location;
 }
 
-const NewModelOptions = (props: NewModelOptionsProps) => {
-  const hasDataAccess = useSelector(() =>
-    getHasDataAccess(props.databases ?? []),
-  );
-  const hasNativeWrite = useSelector(() =>
-    getHasNativeWrite(props.databases ?? []),
-  );
+const NewModelOptions = ({ location }: NewModelOptionsProps) => {
+  const { data } = useListDatabasesQuery();
```

Comment:
We will have redundant requests because of this, e.g. opening `/model/new` in master triggers GET `/api/database` once but in this branch we will call that endpoint twice.

We could probably use RTK-generated APIs directly in the entity framework to avoid them.

Reply:
Nice idea, let me try